src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,198 @@
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "../storage/table.hpp"
+#include "abstract_operator.hpp"
+#include "all_type_variant.hpp"
+#include "resolve_type.hpp"
+#include "storage/dictionary_segment.hpp"
+#include "storage/reference_segment.hpp"
+#include "table_scan.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+TableScan::TableScan(const std::shared_ptr<const AbstractOperator> in, ColumnID column_id, const ScanType scan_type,
+                     const AllTypeVariant search_value)
+    : AbstractOperator(in, nullptr), _column_id(column_id), _scan_type(scan_type), _search_value(search_value) {}
+
+ColumnID TableScan::column_id() const { return _column_id; }
+
+ScanType TableScan::scan_type() const { return _scan_type; }
+
+const AllTypeVariant& TableScan::search_value() const { return _search_value; }
+
+std::shared_ptr<const Table> TableScan::_on_execute() {
+
+  const auto in_table = _input_table_left();
+  auto out_table = std::make_shared<Table>();
```

Comment:
You could move this declaration down to the spot where you first use the `out_table` to improve readability